### PR TITLE
[UBSAN] initialize EcalEBTrigPrimTestAlgo::theMapping_ data member

### LINKED ISSUE #35538: [UBSAN] uninitiaized data member
I am trying to debug the UBSAN errors [a] and I noticed that `EcalEBTrigPrimTestAlgo::theMapping_` is not initialized any where ( at least I can not find its ref any where in the cc file). @cms-sw/l1-l2 @cms-sw/upgrade-l2  any idea how this data member is suppose to initialize?

[a] https://cmssdt.cern.ch/SDT/jenkins-artifacts/ubsan_logs/CMSSW_12_1_X_2021-10-01-2300/
```
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:329:20: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:330:25: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:335:20: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:341:33: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:393:56: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
24 Geometry/EcalMapping/src/EcalElectronicsMapping.cc:394:67: runtime error: member call on misaligned address 0x3d4ccccd3d4ccccd for type 'const struct EcalElectronicsMapping', which requires 8 byte alignment
```

### PR DESCRIPTION
**Pull request by smuzaffar**
EcalEBTrigPrimTestAlgo::theMapping_ data member was never initialized which is causes UBSAN runtime errors mentioned https://github.com/cms-sw/cmssw/issues/35538. This PR proposes to initialize the data member. Local build of this change in UBSAN shows that it fixes the #35538 .
fixes #35538

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by rekovic** [comment]
+1

File: SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBTrigPrimTestAlgo.cc [hunk 1]
```diff
@@ -78,6 +76,7 @@ EcalEBTrigPrimTestAlgo::EcalEBTrigPrimTestAlgo(int nSam, int binofmax, bool tcpF
 //----------------------------------------------------------------------
 void EcalEBTrigPrimTestAlgo::init() {
   // initialise data structures
+  theMapping_ = new EcalElectronicsMapping();
```
Comment (thomreis) [inline SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBTrigPrimTestAlgo.cc:79]: I though we should use smart pointers instead nowadays?

**Comment by smuzaffar** [comment]
@cms-sw/upgrade-l2 can you please review this?

**Pull request merged**